presto-hive/src/main/java/com/facebook/presto/hive/BackgroundHiveSplitLoader.java

```diff
@@ -421,6 +430,58 @@ private void loadPartition(HivePartitionMetadata partition)
         fileIterators.addLast(iterator);
     }
 
+    private void addLzoThriftSplitsToQueue(
+            HiveFileIterator hiveFileIterator,
+            String partitionName,
+            Properties schema,
+            List<HivePartitionKey> partitionKeys,
+            ConnectorSession session,
+            TupleDomain<HiveColumnHandle> effectivePredicate,
+            Map<Integer, HiveType> columnCoercions)
+            throws IOException
+    {
+        if (bucketHandle.isPresent()) {
+            throw new PrestoException(StandardErrorCode.NOT_SUPPORTED, "Bucketed table in ThriftGeneralInputFormat is not yet supported");
+        }
+        List<Iterator<HiveSplit>> iteratorList = new ArrayList<>();
+
+        while (hiveFileIterator.hasNext()) {
+            LocatedFileStatus file = hiveFileIterator.next();
+            if (!ThriftGeneralInputFormat.lzoSuffixFilter.accept(file.getPath())) {
+                continue;
+            }
+
+            Configuration targetConfiguration = hdfsEnvironment.getConfiguration(file.getPath());
+            JobConf targetJob = new JobConf(targetConfiguration);
```

Comment:
Working with Hadoop configs can sometimes be expensive actually, they're notoriously inefficient. If there's a way we can get these configs once outside of the loop we should.